src/rail/estimation/algos/deepdisc.py

```diff
@@ -143,11 +145,41 @@ class DeepDiscInformer(CatInformer):
     def __init__(self, args, comm=None):
         CatInformer.__init__(self, args, comm=comm)
 
+    def inform(self, training_data, metadata):
+        self.set_data('input', training_data)
+        self.set_data('metadata', metadata)
+        self.run()
+        self.finalize()
+        return self.get_handle('model')
+
+    def finalize(self):
+        self.temp_dir.cleanup()
+
     def run(self):
         """
         Train a inception NN on a fraction of the training data
         """
         train_data = self.get_data("input")
+        metadata = self.get_data("metadata")
+
+        # create an iterator here
+        itr = self.input_iterator("input")
+        self.temp_dir = tempfile.TemporaryDirectory()
```

Comment:
@OliviaLynn pointed out an [example](https://stackoverflow.com/questions/3223604/how-do-i-create-a-temporary-directory-in-python) of creating a temporary directory using a context manager to ensure that the temp directory would be cleaned up. Currently we're doing this manually in the finalize() method, however, if there is a runtime exception in the `run` method, we won't correctly clean up the temp directory. 

For the sake of code cleanliness, we should probably do this. One approach would be in the `inform` method:
```
def inform(self, ...):
    with tempfile.TemporaryDirectory as temp_directory_name:
        self.temp_dir = temp_directory_name
        <everything else that is already in inform()>
```

Reply:
working on this in https://github.com/LSSTDESC/rail_deepdisc/issues/18